robolectric/src/main/java/org/robolectric/android/internal/AndroidTestEnvironment.java

```diff
@@ -147,9 +153,23 @@ public void setUpApplicationState(
       loggingInitialized = true;
     }
 
-    if (Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
-      Security.addProvider(new BouncyCastleProvider());
+    SecurityMode.Mode securityMode = configuration.get(SecurityMode.Mode.class);
+    if (securityMode == SecurityMode.Mode.BOUNCY_CASTLE || securityMode == null) {
+
+      if (Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
+        Security.addProvider(new BouncyCastleProvider());
+        Log.e("Bouncy Castle", "Bouncy Castle's if condition");
```

Comment:
nit: Maybe:

`Log.i("Bouncy Castle", "Setting security provider to BouncyCastle");`

Same with the Conscrypt log below as well.

Reply:
done